src/Foreign/Haskell/NonEmpty.agda

```diff
@@ -0,0 +1,38 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- The NonEmpty type which calls out to Haskell via the FFI
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K #-}
+
+module Foreign.Haskell.NonEmpty where
+
+open import Level
+open import Data.List.NonEmpty.Base as Data using (_∷_)
+open import Data.List using (List)
+
+{-# FOREIGN GHC import qualified Data.List.NonEmpty as NE #-}
+
+private
+  variable
+    a : Level
+    A : Set a
+
+------------------------------------------------------------------------
+-- Definition
+
+data NonEmpty (A : Set a) : Set a where
+  _∷_ : A → List A → NonEmpty A
+
+{-# FOREIGN GHC type AgdaNonEmpty l a = NE.NonEmpty a #-}
+{-# COMPILE GHC NonEmpty = data AgdaNonEmpty ((|:)) #-}
+
+------------------------------------------------------------------------
+-- Conversion
+
+toForeign : Data.List⁺ A → NonEmpty A
```

Comment:
I'm not sure we need these conversion functions these days now that we have the `coerce` mechanism.

Reply:
I'd prefer to mirror the other `Foreign` modules, by either adding those functions here too, or removing/deprecating the ones in the other modules. Should I switch to the second option?

Reply:
I think it's the right thing to do, yes. Deprecate it in favour of `coerce`.